### PR TITLE
r/spot_fleet_request - on demand arguments

### LINKED ISSUE #7011: aws_spot_fleet_request OnDemandTargetCapacity
_This issue was originally opened by @deitch as hashicorp/terraform#19907. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Current Terraform Version

```
v0.11.11
```

### Use-cases

AWS Spot Fleets support having a certain minimum number of target capacity as on-demand instances, to ensure a minimum threshold of always-available instances. See [API here](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotFleet.html), [CLI here](https://docs.aws.amazon.com/cli/latest/reference/ec2/request-spot-fleet.html), as well as [this example](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-fleet-examples.html#fleet-config7)

I might have missed it, but it doesn't appear in the parameters for [aws_spot_fleet_request](https://www.terraform.io/docs/providers/aws/r/spot_fleet_request.html)

### References

Not much to add, unless I missed that it exists? 

Thanks!

### LINKED ISSUE #16607: Need ability to specify "on-demand" instance count for aws_spot_fleet_request resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
```
$ terraform -version
Terraform v0.12.17
+ provider.aws v3.2.0
```

I created an AWS launch template via the aws_launch_template resource, and am creating a spot fleet request using the template via the aws_spot_fleet_request resource.

In the AWS console, when creating a spot fleet request using a launch template, I have the option to specify on-demand instances.

However, I don't see that option in the aws_spot_fleet_request resource. So even though I'm able to create the spot fleet request, all of the instances are spot instances, and no on-demand instances. Can we add it please?

I posted [Setting AWS on-demand instances using Terraform aws_spot_fleet_request resource using a launch_template?](https://stackoverflow.com/questions/65152811/setting-aws-on-demand-instances-using-terraform-aws-spot-fleet-request-resource) in Stackoverflow which has the screen shot of the AWS console option I mention.

### New or Affected Resource(s)

* aws_spot_fleet_request

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #7011
Closes #16607

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_spot_fleet_request: add `on_demand_allocation_strategy`, `on_demand_max_total_price`, and `on_demand_target_capacity` arguments
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSpotFleetRequest_onDemand'
--- PASS: TestAccAWSSpotFleetRequest_onDemandTargetCapacity (436.81s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice (282.42s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandAllocationStrategy (349.56s)
```

### CODE REVIEW

File: aws/resource_aws_spot_fleet_request.go [hunk 1]
```diff
@@ -894,6 +913,18 @@ func resourceAwsSpotFleetRequestCreate(d *schema.ResourceData, meta interface{})
 		spotFleetConfig.SpotPrice = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOkExists("on_demand_target_capacity"); ok {
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_spot_fleet_request.go:916]: Yeah, i only used it to get around the fact the `GetOk` doesn't work with zero values (also stated in the doc)
if you can suggest another function ill change this (and for target capacity as well)

**Comment by pschiffe** [comment]
Hello, any updates reg this PR? What is missing?

**Comment by ZillaG** [comment]
I gave this a thumbs up. Can we please merge this? It IS missing functionality.

**Comment by DrFaust92** [comment]
rebased + validations

**Comment by ZillaG** [comment]
What version will this be released in?

**Comment by DrFaust92** [comment]
Rebased + changelog

File: aws/resource_aws_spot_fleet_request.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ewbankkit) [reply aws/resource_aws_spot_fleet_request.go]: I replaced the `GetOkExists` with plain value assignment as these are all `Optional` integer arguments so there will always be a value.

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.
In order to get this merged quickly I pushed a couple of minor changes:

* Removed the EC2 instance type validations - We don't do this for the `aws_instance` resource and given the pace at which AWS introduce new instance types I think it could be limiting for practitioners to be restricted to the values that were listed in the version of the AWS Go SDK that was referenced when the provider was built
* Replaced `GetOkExists` with plain value assignment (see [this comment](https://github.com/hashicorp/terraform-provider-aws/pull/13127#discussion_r653556316))

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSSpotFleetRequest_'                                                                    
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSpotFleetRequest_ -timeout 180m
=== RUN   TestAccAWSSpotFleetRequest_basic
=== PAUSE TestAccAWSSpotFleetRequest_basic
=== RUN   TestAccAWSSpotFleetRequest_tags
=== PAUSE TestAccAWSSpotFleetRequest_tags
=== RUN   TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== PAUSE TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== RUN   TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== RUN   TestAccAWSSpotFleetRequest_onDemandTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_onDemandTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
=== PAUSE TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
=== RUN   TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== PAUSE TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== RUN   TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== PAUSE TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== RUN   TestAccAWSSpotFleetRequest_fleetType
=== PAUSE TestAccAWSSpotFleetRequest_fleetType
=== RUN   TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== PAUSE TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== RUN   TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== PAUSE TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== RUN   TestAccAWSSpotFleetRequest_updateTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_updateTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== PAUSE TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== RUN   TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_withoutSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_withoutSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_diversifiedAllocation
=== PAUSE TestAccAWSSpotFleetRequest_diversifiedAllocation
=== RUN   TestAccAWSSpotFleetRequest_multipleInstancePools
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstancePools
=== RUN   TestAccAWSSpotFleetRequest_withWeightedCapacity
=== PAUSE TestAccAWSSpotFleetRequest_withWeightedCapacity
=== RUN   TestAccAWSSpotFleetRequest_withEBSDisk
=== PAUSE TestAccAWSSpotFleetRequest_withEBSDisk
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
=== RUN   TestAccAWSSpotFleetRequest_withTags
=== PAUSE TestAccAWSSpotFleetRequest_withTags
=== RUN   TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== PAUSE TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== RUN   TestAccAWSSpotFleetRequest_WithELBs
=== PAUSE TestAccAWSSpotFleetRequest_WithELBs
=== RUN   TestAccAWSSpotFleetRequest_WithTargetGroups
=== PAUSE TestAccAWSSpotFleetRequest_WithTargetGroups
=== RUN   TestAccAWSSpotFleetRequest_zero_capacity
=== PAUSE TestAccAWSSpotFleetRequest_zero_capacity
=== RUN   TestAccAWSSpotFleetRequest_CapacityRebalance
=== PAUSE TestAccAWSSpotFleetRequest_CapacityRebalance
=== RUN   TestAccAWSSpotFleetRequest_WithInstanceStoreAmi
    provider_test.go:57: Test fails due to test harness constraints
--- SKIP: TestAccAWSSpotFleetRequest_WithInstanceStoreAmi (0.00s)
=== RUN   TestAccAWSSpotFleetRequest_disappears
=== PAUSE TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_basic
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== CONT  TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== CONT  TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== CONT  TestAccAWSSpotFleetRequest_withEBSDisk
=== CONT  TestAccAWSSpotFleetRequest_withWeightedCapacity
=== CONT  TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== CONT  TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== CONT  TestAccAWSSpotFleetRequest_multipleInstancePools
=== CONT  TestAccAWSSpotFleetRequest_diversifiedAllocation
=== CONT  TestAccAWSSpotFleetRequest_withoutSpotPrice
=== CONT  TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== CONT  TestAccAWSSpotFleetRequest_updateTargetCapacity
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== CONT  TestAccAWSSpotFleetRequest_WithELBs
=== CONT  TestAccAWSSpotFleetRequest_WithTargetGroups
--- PASS: TestAccAWSSpotFleetRequest_placementTenancyAndGroup (78.20s)
=== CONT  TestAccAWSSpotFleetRequest_CapacityRebalance
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (126.34s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
--- PASS: TestAccAWSSpotFleetRequest_onDemandAllocationStrategy (146.20s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3 (146.66s)
=== CONT  TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (161.49s)
=== CONT  TestAccAWSSpotFleetRequest_onDemandTargetCapacity
--- PASS: TestAccAWSSpotFleetRequest_instanceInterruptionBehavior (161.53s)
=== CONT  TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
--- PASS: TestAccAWSSpotFleetRequest_disappears (164.98s)
=== CONT  TestAccAWSSpotFleetRequest_zero_capacity
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId (165.87s)
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
--- PASS: TestAccAWSSpotFleetRequest_withEBSDisk (166.14s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
--- PASS: TestAccAWSSpotFleetRequest_withoutSpotPrice (167.17s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate
--- PASS: TestAccAWSSpotFleetRequest_basic (171.54s)
=== CONT  TestAccAWSSpotFleetRequest_withTags
--- PASS: TestAccAWSSpotFleetRequest_multipleInstancePools (172.07s)
=== CONT  TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId (174.74s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate_multiple
--- PASS: TestAccAWSSpotFleetRequest_diversifiedAllocation (177.65s)
=== CONT  TestAccAWSSpotFleetRequest_iamInstanceProfileArn
--- PASS: TestAccAWSSpotFleetRequest_CapacityRebalance (119.29s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
--- PASS: TestAccAWSSpotFleetRequest_withWeightedCapacity (202.42s)
=== CONT  TestAccAWSSpotFleetRequest_associatePublicIpAddress
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList (206.10s)
=== CONT  TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet (206.69s)
=== CONT  TestAccAWSSpotFleetRequest_tags
--- PASS: TestAccAWSSpotFleetRequest_WithELBs (223.41s)
=== CONT  TestAccAWSSpotFleetRequest_fleetType
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList (124.24s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateWithOverrides (130.30s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice (140.75s)
--- PASS: TestAccAWSSpotFleetRequest_withTags (117.47s)
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion (133.56s)
--- PASS: TestAccAWSSpotFleetRequest_iamInstanceProfileArn (122.39s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate (133.82s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate_multiple (128.04s)
--- PASS: TestAccAWSSpotFleetRequest_WithTargetGroups (303.04s)
--- PASS: TestAccAWSSpotFleetRequest_associatePublicIpAddress (116.43s)
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3 (123.11s)
--- PASS: TestAccAWSSpotFleetRequest_zero_capacity (181.26s)
--- PASS: TestAccAWSSpotFleetRequest_fleetType (124.50s)
--- PASS: TestAccAWSSpotFleetRequest_tags (146.05s)
--- PASS: TestAccAWSSpotFleetRequest_updateTargetCapacity (379.77s)
--- PASS: TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate (225.46s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec (236.91s)
--- PASS: TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy (236.66s)
--- PASS: TestAccAWSSpotFleetRequest_changePriceForcesNewRequest (218.91s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandTargetCapacity (287.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	452.768s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSSpotFleetRequest_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSpotFleetRequest_ -timeout 180m
=== RUN   TestAccAWSSpotFleetRequest_basic
=== PAUSE TestAccAWSSpotFleetRequest_basic
=== RUN   TestAccAWSSpotFleetRequest_tags
=== PAUSE TestAccAWSSpotFleetRequest_tags
=== RUN   TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== PAUSE TestAccAWSSpotFleetRequest_associatePublicIpAddress
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate
=== RUN   TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplate_multiple
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
=== RUN   TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== PAUSE TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== RUN   TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== PAUSE TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
=== RUN   TestAccAWSSpotFleetRequest_onDemandTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_onDemandTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
=== PAUSE TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
=== RUN   TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== PAUSE TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== RUN   TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== PAUSE TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== RUN   TestAccAWSSpotFleetRequest_fleetType
=== PAUSE TestAccAWSSpotFleetRequest_fleetType
=== RUN   TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== PAUSE TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== RUN   TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== PAUSE TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== RUN   TestAccAWSSpotFleetRequest_updateTargetCapacity
=== PAUSE TestAccAWSSpotFleetRequest_updateTargetCapacity
=== RUN   TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== PAUSE TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== PAUSE TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
=== RUN   TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_overriddingSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_withoutSpotPrice
=== PAUSE TestAccAWSSpotFleetRequest_withoutSpotPrice
=== RUN   TestAccAWSSpotFleetRequest_diversifiedAllocation
=== PAUSE TestAccAWSSpotFleetRequest_diversifiedAllocation
=== RUN   TestAccAWSSpotFleetRequest_multipleInstancePools
=== PAUSE TestAccAWSSpotFleetRequest_multipleInstancePools
=== RUN   TestAccAWSSpotFleetRequest_withWeightedCapacity
=== PAUSE TestAccAWSSpotFleetRequest_withWeightedCapacity
=== RUN   TestAccAWSSpotFleetRequest_withEBSDisk
=== PAUSE TestAccAWSSpotFleetRequest_withEBSDisk
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== RUN   TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
=== PAUSE TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
=== RUN   TestAccAWSSpotFleetRequest_withTags
=== PAUSE TestAccAWSSpotFleetRequest_withTags
=== RUN   TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== PAUSE TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== RUN   TestAccAWSSpotFleetRequest_WithELBs
=== PAUSE TestAccAWSSpotFleetRequest_WithELBs
=== RUN   TestAccAWSSpotFleetRequest_WithTargetGroups
=== PAUSE TestAccAWSSpotFleetRequest_WithTargetGroups
=== RUN   TestAccAWSSpotFleetRequest_zero_capacity
=== PAUSE TestAccAWSSpotFleetRequest_zero_capacity
=== RUN   TestAccAWSSpotFleetRequest_CapacityRebalance
=== PAUSE TestAccAWSSpotFleetRequest_CapacityRebalance
=== RUN   TestAccAWSSpotFleetRequest_WithInstanceStoreAmi
    provider_test.go:57: Test fails due to test harness constraints
--- SKIP: TestAccAWSSpotFleetRequest_WithInstanceStoreAmi (0.00s)
=== RUN   TestAccAWSSpotFleetRequest_disappears
=== PAUSE TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_basic
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
=== CONT  TestAccAWSSpotFleetRequest_onDemandAllocationStrategy
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
=== CONT  TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
=== CONT  TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy
=== CONT  TestAccAWSSpotFleetRequest_updateTargetCapacity
=== CONT  TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
=== CONT  TestAccAWSSpotFleetRequest_iamInstanceProfileArn
=== CONT  TestAccAWSSpotFleetRequest_fleetType
=== CONT  TestAccAWSSpotFleetRequest_instanceInterruptionBehavior
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3
=== CONT  TestAccAWSSpotFleetRequest_disappears
=== CONT  TestAccAWSSpotFleetRequest_CapacityRebalance
=== CONT  TestAccAWSSpotFleetRequest_zero_capacity
=== CONT  TestAccAWSSpotFleetRequest_WithTargetGroups
=== CONT  TestAccAWSSpotFleetRequest_WithELBs
=== CONT  TestAccAWSSpotFleetRequest_placementTenancyAndGroup
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateWithOverrides
--- PASS: TestAccAWSSpotFleetRequest_placementTenancyAndGroup (80.07s)
=== CONT  TestAccAWSSpotFleetRequest_withTags
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion (139.77s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3
--- PASS: TestAccAWSSpotFleetRequest_CapacityRebalance (141.20s)
=== CONT  TestAccAWSSpotFleetRequest_onDemandTargetCapacity
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (144.31s)
=== CONT  TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice
--- PASS: TestAccAWSSpotFleetRequest_basic (150.50s)
=== CONT  TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate
--- PASS: TestAccAWSSpotFleetRequest_iamInstanceProfileArn (155.98s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList (155.98s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplate_multiple
--- PASS: TestAccAWSSpotFleetRequest_onDemandAllocationStrategy (156.95s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstancePools
--- PASS: TestAccAWSSpotFleetRequest_disappears (158.90s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateWithOverrides (162.22s)
=== CONT  TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId
--- PASS: TestAccAWSSpotFleetRequest_instanceInterruptionBehavior (166.09s)
=== CONT  TestAccAWSSpotFleetRequest_withEBSDisk
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDeviceGp3 (167.11s)
=== CONT  TestAccAWSSpotFleetRequest_withWeightedCapacity
--- PASS: TestAccAWSSpotFleetRequest_fleetType (175.63s)
=== CONT  TestAccAWSSpotFleetRequest_associatePublicIpAddress
--- PASS: TestAccAWSSpotFleetRequest_withTags (121.84s)
=== CONT  TestAccAWSSpotFleetRequest_tags
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList (203.77s)
=== CONT  TestAccAWSSpotFleetRequest_withoutSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_WithELBs (206.74s)
=== CONT  TestAccAWSSpotFleetRequest_diversifiedAllocation
--- PASS: TestAccAWSSpotFleetRequest_zero_capacity (260.02s)
=== CONT  TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec
=== CONT  TestAccAWSSpotFleetRequest_overriddingSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_changePriceForcesNewRequest (262.31s)
--- PASS: TestAccAWSSpotFleetRequest_updateExcessCapacityTerminationPolicy (265.76s)
=== CONT  TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDeviceGp3 (130.68s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandMaxTotalPrice (127.50s)
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_RootBlockDevice_KmsKeyId (126.86s)
--- PASS: TestAccAWSSpotFleetRequest_LaunchSpecification_EbsBlockDevice_KmsKeyId (127.45s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate_multiple (134.77s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplate (136.16s)
--- PASS: TestAccAWSSpotFleetRequest_withEBSDisk (126.40s)
--- PASS: TestAccAWSSpotFleetRequest_multipleInstancePools (141.44s)
--- PASS: TestAccAWSSpotFleetRequest_associatePublicIpAddress (123.51s)
--- PASS: TestAccAWSSpotFleetRequest_withoutSpotPrice (116.33s)
--- PASS: TestAccAWSSpotFleetRequest_WithTargetGroups (321.33s)
--- PASS: TestAccAWSSpotFleetRequest_diversifiedAllocation (128.09s)
--- PASS: TestAccAWSSpotFleetRequest_withWeightedCapacity (180.61s)
--- PASS: TestAccAWSSpotFleetRequest_tags (150.96s)
--- PASS: TestAccAWSSpotFleetRequest_launchSpecToLaunchTemplate (221.51s)
--- PASS: TestAccAWSSpotFleetRequest_updateTargetCapacity (388.30s)
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (132.18s)
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet (151.23s)
--- PASS: TestAccAWSSpotFleetRequest_onDemandTargetCapacity (293.73s)
--- PASS: TestAccAWSSpotFleetRequest_launchTemplateToLaunchSpec (227.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	491.847s
```

**Pull request merged**